services/common/src/components/reports/reportSubmissionSlice.ts

```diff
@@ -0,0 +1,93 @@
+import { hideLoading, showLoading } from "react-redux-loading-bar";
+import { ENVIRONMENT, IMineReportSubmission } from "../..";
+import { createAppSlice } from "@mds/common/redux/createAppSlice";
+import CustomAxios from "@mds/common/redux/customAxios";
+import * as API from "@mds/common/constants/API";
+import { RootState } from "@mds/common/redux/rootState";
+
+interface ReportSubmissionState {
+  reportSubmission: IMineReportSubmission;
+  mineReportGuid: string;
+}
+
+const initialState: ReportSubmissionState = {
+  reportSubmission: null,
+  mineReportGuid: null,
+};
+
+const createRequestHeader = REQUEST_HEADER.createRequestHeader;
+
+// could probably make this generic
+// the stack doesn't print nicely with action.error, action.error.stack outputs the \n properly
+const rejectHandler = (action) => {
+  console.log(action.error);
+  console.log(action.error.stack);
+};
+
+const submissionSlice = createAppSlice({
+  name: "reportSubmissions",
+  initialState: initialState,
+  reducers: (create) => ({
+    fetchLatestReportSubmission: create.asyncThunk(
+      async (payload: { mine_report_guid: string }, thunkApi) => {
+        const headers = createRequestHeader();
+        thunkApi.dispatch(showLoading());
+        const resp = await CustomAxios({
+          errorToastMessage: "Failed to load report submission",
+        }).get(`${ENVIRONMENT.apiUrl}${API.MINE_REPORT_SUBMISSIONS(payload, true)}`, headers);
+        thunkApi.dispatch(hideLoading());
+        return resp.data;
+      },
+      {
+        fulfilled: (state, action) => {
```

Comment:
the fulfilled/rejected config (also available: pending, settled- I just didn't need them) is pretty neat, but it can't dispatch actions and it can't trigger a toast notification. It can change the slice state though. That's why I gave a success toast message param to custom axios.